connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroOptions.scala

```diff
@@ -139,11 +141,19 @@ private[sql] object AvroOptions {
     new AvroOptions(CaseInsensitiveMap(parameters), hadoopConf)
   }
 
-  val ignoreExtensionKey = "ignoreExtension"
+  val IGNORE_EXTENSION_KEY = Value("ignoreExtension")
```

Comment:
Let's just expose a `Seq[String]` that contains these options. Another option is to create a trait like:
```
trait FormatOptions {
  // Use reflection to get list of all fields that are strings
}
```

Then you add a unit test that hardcodes how many parameters are found.